src/app/modules/dealers/services/dealers.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, Inject } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { Dealers } from '../models/dealers.entity';
+import { MysqlEscape } from '../../common/services/mysql-escape.service';
+import { countryCode } from '../../common/enums/country-code.enum';
+
+@Component()
+export class DealersService {
+
+    constructor(@Inject('DealersRepositoryToken') private readonly dealersRepository: Repository<Dealers>,
+                private mysqlEscape: MysqlEscape) {}
+
+    async findAll(): Promise<Dealers[]> {
+        return await this.dealersRepository.find();
+    }
+
+    async findAllResidential(): Promise<Dealers[]> {
+        return await this.dealersRepository.find({countryCode: countryCode.RESIDENTIAL_COUNTRY});
+    }
+
+    async residentialState(state): Promise<Dealers[]> {
+        state = this.mysqlEscape.mysqlStringEscape(state);
+        return await this.dealersRepository.find({countryCode: countryCode.RESIDENTIAL_COUNTRY, locations: countryCode.STATE + state});
+    }
+
+    async residentialStateRep(state, dealerRep): Promise<Dealers[]> {
+        state = this.mysqlEscape.mysqlStringEscape(state);
+        let isRep = null;
+        if (dealerRep === 'dealer') {
```

Comment:
this is another magic string I missed on the first go. Maybe an enum SellerType or something with dealer and rep values. the strings 'dealer' and 'rep' are significant and should be centralized should we need to change them.